include/ringmesh/mesh/cartesian_grid.h

```diff
@@ -271,8 +300,84 @@ namespace RINGMesh
 
         GEO::AttributesManager attributes_manager_;
     };
+    ALIAS_2D_AND_3D( CartesianGridBase );
+
+    template < index_t DIMENSION >
+    class CartesianGrid final : public CartesianGridBase< DIMENSION >
+    {
+        friend class CartesianGridBuilder< DIMENSION >;
+    };
     ALIAS_2D_AND_3D( CartesianGrid );
 
+    template <>
+    class CartesianGrid< 3 > final : public CartesianGridBase< 3 >
+    {
+        friend class CartesianGridBuilder< 3 >;
+
+    public:
+        CartesianGrid( ivec3 nb_cells_in_each_direction,
+            ReferenceFrame3D vec_cartesian_axis )
+            : CartesianGridBase(
+                  nb_cells_in_each_direction, vec_cartesian_axis ),
+              grid_cage_()
+        {
+            for( auto i : range( 3 ) )
+            {
+                vec3 origin1 = cartesian_frame_.origin();
+                vec3 vector1 = cartesian_frame_[i];
+                grid_cage_.emplace_back( Geometry::Plane{ vector1, origin1 } );
+                vec3 origin2 = cartesian_frame_.origin()
+                               + cartesian_frame_[0] * nb_cells_axis( 0 )
+                               + cartesian_frame_[1] * nb_cells_axis( 1 )
+                               + cartesian_frame_[2] * nb_cells_axis( 2 );
+                vec3 vector2 = cartesian_frame_[i];
+                grid_cage_.emplace_back( Geometry::Plane{ vector2, origin2 } );
+            }
+        }
+
+        const std::vector< Geometry::Plane >& grid_cage() const
+        {
+            return grid_cage_;
+        }
+
+    private:
+        std::vector< Geometry::Plane > grid_cage_;
```

Comment:
Do you have any convention on the order of the grid cage planes?

Reply:
Yes, first the 2 with a normal to the first axis of the grid, then the 2 normal to the second axis, and then the 2 normal to the third axis. Each time, the first of the 2 planes is the one which contains the origin of the grid and the second is the one that contains the point at the opposite of the diagonal of the grid. Not sure it's really useful to know it, but they are written that way. Should I write a commentary on it ?

Reply:
Either the order is important and so write a quick comment as you did in your answer. Or it is not really important and I suggest you can use a set instead of a vector